src/Lean/Meta/Tactic/SplitIf.lean

```diff
@@ -56,23 +56,24 @@ def discharge? (useDecide := false) : Simp.Discharge := fun prop => do
          else
            return none
 
-/-- Return the condition of an `if` expression to case split. -/
-partial def findIfToSplit? (e : Expr) : Option Expr :=
+/-- Return the condition (and decidability instance) of an `if` expression to case split. -/
+partial def findIfToSplit? (e : Expr) : Option (Expr × Expr) :=
   if let some iteApp := e.find? fun e => (e.isIte || e.isDIte) && !(e.getArg! 1 5).hasLooseBVars then
```

Comment:
Do we need `!(e.getArg! 2 5).hasLooseBVars` as well? Do you know how to write a test to check?

Reply:
I wouldn't know how to reliably check whether it's needed or not. On the other hand, we might *only* need `!(e.getArg! 2 5).hasLooseBVars` because the proposition is a subterm of the decidable instance.

Reply:
We do. Here's a contrived example:
```lean
example (c : Prop) :
    (fun [Decidable c] => if c then 0 else 0) = (fun _ => 0) := by
  split <;> rfl
```

Reply:
Note that adding `!(e.getArg! 2 5).hasLooseBVars` will make `split` fail to split the if statement in this example. If we wanted to retain the original behavior, we could say that if `(e.getArg! 2 5).hasLooseBVars` then it just returns `none` for that component (and hence it would use classical logic to split in that case).